handlers.go

```diff
@@ -35,3 +41,99 @@ func (s *server) handleHelloWord() http.HandlerFunc {
 		respond(w, r, res, http.StatusOK)
 	}
 }
+
+func (s *server) handleUserSignUp() func(w http.ResponseWriter, r *http.Request) {
+
+	type response struct {
+		Message string `json:"message"`
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		decoder := json.NewDecoder(r.Body)
+
+		// var data map[string]interface{}
+		var newUser user
+		err := decoder.Decode(&newUser)
+
+		if err != nil {
+			respondErr(w, r, err, http.StatusInternalServerError)
+			return
+		}
+
+		passEncrypted, _ := bcrypt.GenerateFromPassword([]byte(newUser.Password), 14)
+
+		_, err = s.db.Exec("INSERT INTO Users (name, password, email) VALUES ($1, $2, $3)",
+			newUser.Name,
+			passEncrypted,
+			newUser.Email,
+		)
+
+		if err != nil {
+			respondErr(w, r, err, http.StatusInternalServerError)
+			return
+		}
+
+		resp := response{"User registered"}
+		respond(w, r, resp, http.StatusOK)
+	}
+}
+
+func (s *server) handleUserSignIn() func(w http.ResponseWriter, r *http.Request) {
+
+	type responseToken struct {
+		Message string `json:"token"`
+	}
+
+	type response struct {
+		Message string `json:"message"`
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		decoder := json.NewDecoder(r.Body)
+
+		// var data map[string]interface{}
+		var newUser user
```

Comment:
Nitpick: Perhaps it is better to change the naming here instead of `newUser`, since during sign in, the users technically will not be a "new user".